game/src/engine/physics/PhysicsService.cpp

```diff
@@ -97,10 +99,13 @@ void PhysicsService::OnSceneLoaded(Scene& scene)
 
 void PhysicsService::OnUpdate()
 {
-    const Timestep& delta = GetApp().GetDeltaTime();
-    time_accumulator_ += delta;
-
-    StepPhysics();
+    if (!game_state_service_->GetDisplayPauseBoolean())
+    {
+        const Timestep& delta = GetApp().GetDeltaTime();
+        time_accumulator_ += delta;
+        
+        StepPhysics();
+    }
```

Comment:
Rather than having the PhysicsService ask the GameStateService whether to pause or not, could we flip it around? So like, have the PhysicsService have a Pause/Unpause func, and call that from within GameStateService instead, and then remove any references to GameStateService from this file

Reason being, I think it'd be better to keep the things that are strictly engine separate from the game logic. i.e. nothing within the `engine/` folder should ever need to include anything from the `game/` folder, only the other way around